### DISCUSSION TITLE
Open web project on debug

### DISCUSSION THREAD

**Discussion by mip1983** [microsoft/aspire#4414] [Q&A]
When you run your aspire project from VS2022 with the AppHost as startup project it launches the Aspire dashboard in a browser as you'd want/expect. Is there any way to tell it to also launch one (or more) of the AppHosts hosted web projects in another tab (or window) at the same time automatically? Like multiple startup projects but running them as hosted by aspire.

I have two aspire projects, one is Blazor server, and the other is mixed Blazor web (server and wasm), and the wasm project does automatically open it's own browser for the web project, I think because it needs some special instance of the browser to attach debugging to the client side stuff. Similar behavior on the server project would be nice to enable if possible.

**Comment by davidfowl** [comment]
> Is there any way to tell it to also launch one (or more) of the AppHosts hosted web projects in another tab (or window) at the same time automatically? Like multiple startup projects but running them as hosted by aspire.

No there's no support for this.

**Comment by Makajda** [comment]
https://learn.microsoft.com/ru-ru/dotnet/aspire/fundamentals/launch-profiles#control-launch-profile-selection

**Comment by waf** [comment]
I've worked around this by just launching the browser in the AppHost.cs. For my particular use case, this works well:

```csharp

// projects registered above

var app = builder.Build();

app.Services
    .GetRequiredService<IHostApplicationLifetime>().ApplicationStarted
    .Register(() => Process.Start(
        new ProcessStartInfo("https://some-url-here/") { UseShellExecute = true }
    ));

app.Run();
```
